### ISSUE TITLE
Support IdentityStore

### ISSUE THREAD

**Issue by yevgenypats** [cloudquery/cloudquery#1418]
IdentityStore is a key service in AWS at scale management. Add support for retrieving users and groups.

https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/identitystore

**Comment by jsonpr** [comment]
AWS released APIs to AWS Identity Center (formerly known as AWS SSO) that will permit us to pull the proper data from AWS Identity Center.

https://aws.amazon.com/blogs/security/announcing-new-aws-iam-identity-center-apis-to-manage-users-and-groups-at-scale/

**Comment by jhudson10x** [comment]
I started working on this today. I anticipate having a PR open by the end of the week.

**Comment by yevgenypats** [comment]
@jhudson10x Nice! Let us know if you have any questions/need help.

**Comment by jhudson10x** [comment]
I've got the bulk of the code written but I'm running into an issue. The plugin compiles but the plugin sdk is not invoking the new table resolver. I can see that the table is registered because the table name validation step in the plugin is successful. I don't know why https://github.com/cloudquery/plugin-sdk/blob/051e24710b64672b4fa4eda1261e2558859cbc75/schema/table.go#L160 is not being called.

**Comment by hermanschaaf** [comment]
@jhudson10x Were you able to figure it out? Your code looks good and I ran it locally to confirm that the resolver is now being called (I see the `*******FETCH********` and `*******RESPONSE********` print lines). One thing to note maybe is just that when you're developing on a plugin, the console output will go to the plugin process's terminal, not the cloudquery.log file.

Let us know if you need any further help!

**Comment by jhudson10x** [comment]
Thank you @hermanschaaf. Yes, I was able to get past the issue. `cq-ben` in discord (not sure what their handle is here) pointed out that there is no `ssoadmin` key in `plugins/source/aws/client/data/partition_service_region.json`. I changed the region multiplexer to use the `identitystore` key and am unblocked now.

I have two more (non-blocking) questions now:

1. The service has to be configured in exactly one region. Right now it queries each of the 13 regions listed in identitystore, 12 of which always fail with a 400. Is there an elegant way to deal with this? I am not aware of one.
2. The service can't be configured in every AWS region, but it can be configured in more than just the 13 regions defined by the `identitystore` key. Should I add a new `ssoadmin` key with the correct list of regions?

**Comment by bbernays** [comment]
> Thank you @hermanschaaf. Yes, I was able to get past the issue. `cq-ben` in discord (not sure what their handle is here) 
Hi there! This is my Github handle! 
> I have two more (non-blocking) questions now:
> 
> 1. The service has to be configured in exactly one region. Right now it queries each of the 13 regions listed in identitystore, 12 of which always fail with a 400. Is there an elegant way to deal with this? I am not aware of one.
There is not an elegant way of doing it other than checking all regions that it could be setup in
> 2. The service can't be configured in every AWS region, but it can be configured in more than just the 13 regions defined by the `identitystore` key. Should I add a new `ssoadmin` key with the correct list of regions?
Yes, if the service doesn't exist in the json file (or the source file) then feel free to add in an entry

**Comment by cqgaurav** [comment]
Hey Andrii, moving to Ben feel free to grab again if you started working on it.

**Comment by jhudson10x** [comment]
This is ready for review #3005.

**Comment by jhudson10x** [comment]
It is still missing some key functionality from `ssoadmin` service, but per @erezrokah suggestion we can go ahead and merge this in now and I will follow up with another PR to add the permission sets/accounts piece. This PR has the pieces needed to make `identitystore` service work (ssoadmin.instances + identitystore.[users/groups/group_memberships])
